ffi/capi/src/casemap.rs

```diff
@@ -67,14 +67,12 @@ pub mod ffi {
         #[diplomat::rust_link(icu::casemap::CaseMapper::lowercase_to_string, FnInStruct, hidden)]
         pub fn lowercase(
             &self,
-            s: &str,
+            s: &DiplomatStr,
             locale: &ICU4XLocale,
             write: &mut DiplomatWriteable,
         ) -> Result<(), ICU4XError> {
-            // #2520
-            // In the future we should be able to make assumptions based on backend
```

Comment:
For Diplomat frontends that actually support UTF-8 (like C++20, Swift, Golang, ...) we still want a way to not re-run UTF-8 validation. Is that still in the plan?

Reply:
This is also true for frontends like JavaScript that use a TextEncoder before giving the strings to ICU4X.

Reply:
@sffc Yes, that will be the `str` type.

In the case of JS we should be using the utf-16 endpoint anyway, so JS won't make use of this